packages/agreement-process/test/agreement.integration.test.ts

```diff
@@ -0,0 +1,713 @@
+      id: agreementId,
+      eserviceId: expectedEserviceId,
+      descriptorId: expectedDescriptorId,
+      producerId: expectedProducerId,
+      consumerId: expectedConsumerId,
+      state: toAgreementStateV1(agreementState.draft),
+      verifiedAttributes: [],
+      certifiedAttributes: [],
+      declaredAttributes: [],
+      consumerDocuments: [],
+      stamps: {},
+      createdAt: expect.any(BigInt),
+    });
+
+    return actualAgreement;
+  };
+
+  beforeAll(async () => {
+    postgreSqlContainer = await startPostgresDBContainer(config);
+    mongodbContainer = await startMongoDBContainer(config);
+
+    config.eventStoreDbPort = postgreSqlContainer.getMappedPort(
+      TEST_POSTGRES_DB_PORT
+    );
+    config.readModelDbPort = mongodbContainer.getMappedPort(TETS_MONGO_DB_PORT);
+
+    const readModelRepository = ReadModelRepository.init(config);
+    agreements = readModelRepository.agreements;
+    eservices = readModelRepository.eservices;
+    tenants = readModelRepository.tenants;
+
+    readModelService = readModelServiceBuilder(readModelRepository);
+    const eserviceQuery = eserviceQueryBuilder(readModelService);
+    const agreementQuery = agreementQueryBuilder(readModelService);
+    const tenantQuery = tenantQueryBuilder(readModelService);
+    const attributeQuery = attributeQueryBuilder(readModelService);
+
+    postgresDB = initDB({
+      username: config.eventStoreDbUsername,
+      password: config.eventStoreDbPassword,
+      host: config.eventStoreDbHost,
+      port: config.eventStoreDbPort,
+      database: config.eventStoreDbName,
+      schema: config.eventStoreDbSchema,
+      useSSL: config.eventStoreDbUseSSL,
+    });
+
+    if (!postgresDB) {
+      logger.error("postgresDB is undefined!!");
+    }
+
+    agreementService = agreementServiceBuilder(
+      postgresDB,
+      agreementQuery,
+      tenantQuery,
+      eserviceQuery,
+      attributeQuery
+    );
+  });
+
+  afterEach(async () => {
+    await agreements.deleteMany({});
+    await eservices.deleteMany({});
+    await tenants.deleteMany({});
+
+    await postgresDB.none("TRUNCATE TABLE agreement.events RESTART IDENTITY");
+    await postgresDB.none("TRUNCATE TABLE catalog.events RESTART IDENTITY");
+  });
+
+  afterAll(async () => {
+    await postgreSqlContainer.stop();
+    await mongodbContainer.stop();
+  });
+
+  describe("createAgreement (success cases)", () => {
+    it("succeeds when EService Producer and Agreement Consumer are the same, even on unmet attributes", async () => {
+      const authData = getRandomAuthData();
+      const eserviceId = generateId<EServiceId>();
+      const descriptorId = generateId<DescriptorId>();
+      const attributeId = generateId<AttributeId>();
+
+      const descriptor = buildDescriptorPublished(descriptorId, [
+        [buildEServiceAttribute(attributeId)],
+      ]);
+      const eservice = buildEService(
+        eserviceId,
+        unsafeBrandId<TenantId>(authData.organizationId),
```

Comment:
Minor: maybe `unsafeBrandId` is not needed? `AuthData['organizationId']` should already be a `TenantId`